### ISSUE TITLE
Cannot find the ACS URL to configure in IDP - SAML Authentication

### ISSUE THREAD

**Issue by Shashank-Digicert** [sqlpad/sqlpad#1053]
I am trying to integrate sqlpad with Okta for SAML authentication. 
I am using a Docker image from DockerHub and passing in the Environment variables as follows:

SQLPAD_SAML_AUTH_CONTEXT = "urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport"

SQLPAD_SAML_CALLBACK_URL = "https://--sqlpad.env.zone.org.com--/login/callback"

SQLPAD_SAML_CERT = "--Server from Okta--"

SQLPAD_SAML_ENTRY_POINT = "--SSO URL from OKTA--"

SQLPAD_SAML_ISSUER = "http://www.okta.com/--ID--"

SQLPAD_SAML_LINK_HTML = "Sign in with SSO"

SQLPAD_SAML_AUTO_SIGN_UP = "false"

SQLPAD_SAML_DEFAULT_ROLE = "editor"

PUBLIC_URL = "http://localhost"

SQLPAD_USERPASS_AUTH_DISABLED = true

I am not able to find the ACS URL needed to be configured in Okta. Am I doing something wrong ?

**Comment by Shashank-Digicert** [comment]
As I haven't got any response and I kept digging in the code for what was wrong. I found this in the /usr/app/auth-strategies/saml.js file
````
 async function (req, p, done) {
          const email =
            p[
              'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'
            ];

          // If email is not provided - no auth
          if (!email) {
            return done(null, false);
          }
````
Line # 26 email is assigned with "p['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'];", where the url acts as ID, which doesn't exist in the object we are getting from the IDP. So, printed out the object, for us it was nameID which had the email address from the IDP after authentication. I replaced it with "p['nameID']" and it started working.

Hope this will helpful for someone trying to use sqlpad using SAML Auth.

**Comment by bhicks-adaptive** [comment]
@Shashank-Digicert what'd you use for the audience restriction (aka entity id) on the Okta side?

**Comment by Shashank-Digicert** [comment]
> @Shashank-Digicert what'd you use for the audience restriction (aka entity id) on the Okta side?

I used `sqlpad` but it can be anything as the app doesn't check this value, but on Okta side it is required.

And we have changed the above implementation a little to avoid changing the code in sqlpad. We created an attribute statement with `Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress`, `Name format: Unspecified` and `Value: user.email` on okta side which also fixes the issue.
